parlacards/serializers/cards.py

```diff
@@ -1356,17 +1356,25 @@ def get_results(self, obj):
 
     def to_representation(self, instance):
         parent_data = super().to_representation(instance)
-
         # instance is the mandate
-        # TODO: filter by mandate
-        votes = Vote.objects.filter(timestamp__lte=self.context['date']).order_by('-timestamp')
-
-        # TODO: maybe lemmatize?, maybe search by each word separately?
-        if text := self.context['GET'].get('text', None):
-            votes = votes.filter(motion__text__icontains=text)
 
         requested_page, requested_per_page = parse_pagination_query_params(self.context['GET'])
-        paginator = Paginator(votes, requested_per_page)
+
+        if text := self.context['GET'].get('text', None):
+            solr_params = parse_search_query_params(self.context['GET'])
+            solr_params['mandate'] = instance.description
+            paginator = SolrPaginator(
+                solr_params,
+                requested_per_page,
+                document_type='vote'
+            )
+        else:
+            votes = Vote.objects.filter(
+                timestamp__lte=self.context['date'],
+                motion__session__mandate=instance
+            ).order_by('-timestamp')
+            paginator = Paginator(votes, requested_per_page)
```

Comment:
A je kak razlog, da tu kličeš solr samo ko je treba filtrirat? Pri govorih se vedno kliče solr tud ko ni search querija.

Reply:
Hja pomojem ja. Ker v vsakem primeru morem še pol bazo klicat. Iz solr-ja bi pridobil samo id-je. To pa lahko tudi direkt na bazi nardim. @muki kaj misliš ti?
Ker druga opcija je da na solr shranim cel objekt serializiranega vote-a/zakona sam pol bi blo trebalo updejtat solr ko se kej spremeni objekt. Naredu sem bol po principu kot je blo narjeno za speech-e.

Reply:
jaz bi pustil tako, samo po moje to rabi en komentar, pa todo, da se tudi pri govorih to uredi. sem dodal v trello

https://trello.com/c/r85rjgXM/197-zrihtaj-da-se-solr-kli%C4%8De-samo-ko-je-treba-recimo-govori